Tests/SwiftCurrent_SwiftUITests/SwiftCurrent_SwiftUITests.swift

```diff
@@ -0,0 +1,376 @@
+        }
+
+        wait(for: [expectOnAbandon, expectViewLoaded], timeout: 0.3)
+    }
+
+    func testWorkflowViewCanHaveMultipleOnAbandonCallbacks() throws {
+        struct FR1: View, FlowRepresentable, Inspectable {
+            var _workflowPointer: AnyFlowRepresentable?
+            var body: some View { Text("FR1 type") }
+        }
+        let isPresented = Binding(wrappedValue: true)
+        let expectOnAbandon1 = expectation(description: "OnAbandon1 called")
+        let expectOnAbandon2 = expectation(description: "OnAbandon2 called")
+        let expectViewLoaded = ViewHosting.loadView(
+            WorkflowView(isPresented: isPresented)
+                .thenProceed(with: WorkflowItem(FR1.self))
+                .onAbandon {
+            XCTAssertFalse(isPresented.wrappedValue)
+            expectOnAbandon1.fulfill()
+        }.onAbandon {
+            XCTAssertFalse(isPresented.wrappedValue)
+            expectOnAbandon2.fulfill()
+        }).inspection.inspect { viewUnderTest in
+            XCTAssertNoThrow(try viewUnderTest.find(FR1.self).actualView().workflow?.abandon())
+            XCTAssertThrowsError(try viewUnderTest.find(FR1.self))
+        }
+
+        wait(for: [expectOnAbandon1, expectOnAbandon2, expectViewLoaded], timeout: 0.3)
+    }
+
+    func testWorkflowViewCanHaveModifiers() throws {
+        struct FR1: View, FlowRepresentable, Inspectable {
+            var _workflowPointer: AnyFlowRepresentable?
+            var body: some View { Text("FR1 type") }
+
+            func customModifier() -> Self { self }
+        }
+
+        let expectViewLoaded = ViewHosting.loadView(
+            WorkflowView(isPresented: .constant(true))
+                .thenProceed(with: WorkflowItem(FR1.self)
+                                .applyModifiers { $0.customModifier().background(Color.blue) })).inspection.inspect { viewUnderTest in
+            XCTAssertNoThrow(try viewUnderTest.vStack().anyView(0).view(FR1.self).background())
+        }
+
+        wait(for: [expectViewLoaded], timeout: 0.3)
+    }
+
+    func testWorkflowViewRemovesRemnantsAfterWorkflowDisappears() throws {
+        struct FR1: View, FlowRepresentable, Inspectable {
+            var _workflowPointer: AnyFlowRepresentable?
+            var body: some View { Text("FR1 type") }
+
+            func customModifier() -> Self { self }
+        }
+
+        let expectViewLoaded = ViewHosting.loadView(
+            WorkflowView(isPresented: .constant(true))
+                .thenProceed(with: WorkflowItem(FR1.self)
+                                .applyModifiers { $0.customModifier().background(Color.blue) })).inspection.inspect { viewUnderTest in
+            XCTAssertNoThrow(try viewUnderTest.vStack().callOnDisappear())
+            XCTAssertNoThrow(try viewUnderTest.vStack().anyView(0).emptyView())
+        }
+
+        wait(for: [expectViewLoaded], timeout: 0.3)
+    }
+
+    func testWorkflowOnlyLaunchesOnce() throws {
```

Comment:
It's a slight thing to think about, this one makes sense.  We should only launch the workflow once when the view is visible so that they don't lose their place... but if the view disappears then re-appears (toggling isPresented) then the view will need to restart, correct?

We may need a test for that scenario and maybe some quick research on scrolling a view out of view and back.